addons/mail/static/tests/owl/hooks/use_store_compare_keys_tests.js

```diff
@@ -0,0 +1,143 @@
+            });
+            this.env.store = this.store;
+        };
+    },
+    afterEach() {
+        this.env = undefined;
+        this.store = undefined;
+    }
+});
+
+
+QUnit.test("compare keys, no depth, primitives", async function (assert) {
+    assert.expect(7);
+    this.start();
+    let count = 0;
+    this.store.state.obj = {
+        subObj1: 'a',
+        subObj2: 'b',
+        use1: true,
+    };
+    class MyComponent extends Component {
+        constructor() {
+            super(...arguments);
+            this.storeProps = useStoreCompareKeys((state, props) => {
+                return {
+                    res: state.obj.use1 ? state.obj.subObj1 : state.obj.subObj2,
+                };
+            }, {
+                onUpdate: () => {
+                    count++;
+                },
+            });
+        }
+    }
+    MyComponent.env = this.env;
+    MyComponent.template = xml`<div t-esc="storeProps.res"/>`;
+
+    const fixture = document.querySelector('#qunit-fixture');
+
+    const myComponent = new MyComponent();
+    await myComponent.mount(fixture);
+    assert.strictEqual(count, 0, 'initial state');
+    assert.strictEqual(fixture.textContent, 'a', 'content of subObj1');
+
+    this.store.state.obj.use1 = false;
+    await afterNextRender();
+    assert.strictEqual(count, 1, 'updated once after change');
+    assert.strictEqual(fixture.textContent, 'b', 'content of subObj2');
+
+    this.store.state.obj.subObj2 = 'b';
+    // there must be no render here
+    await nextAnimationFrame();
+    assert.strictEqual(count, 1, 'no actual change, no update');
+
+    this.store.state.obj.subObj2 = 'd';
+    await afterNextRender();
+    assert.strictEqual(count, 2, 'change, update');
+    assert.strictEqual(fixture.textContent, 'd', 'updated content of subObj2');
+
+    myComponent.destroy();
+});
+
+QUnit.test("compare keys, depth 1, proxy", async function (assert) {
+    assert.expect(8);
+    this.start();
+    let count = 0;
+    this.store.state.obj = {
+        subObj1: { a: 'a' },
+        subObj2: { a: 'b' },
+        use1: true,
+    };
+    class MyComponent extends Component {
+        constructor() {
+            super(...arguments);
+            this.storeProps = useStoreCompareKeys((state, props) => {
+                return {
+                    array: [state.obj.use1 ? state.obj.subObj1 : state.obj.subObj2],
+                };
+            }, {
+                compareDepth: {
+                    array: 1,
+                },
+                onUpdate: () => {
+                    count++;
+                },
+            });
+        }
+    }
+    MyComponent.env = this.env;
+    MyComponent.template = xml`<div t-esc="storeProps.array[0].a"/>`;
+
+    const fixture = document.querySelector('#qunit-fixture');
+
+    const myComponent = new MyComponent();
+    await myComponent.mount(fixture);
+    assert.strictEqual(count, 1, 'updated once, initial state when comparing proxy');
```

Comment:
doesn't really make sense that counter is 1 here and 0 in previous test.

Reply:
fixed by giving init state to store